### DISCUSSION TITLE
String functions on sqlite field

### DISCUSSION THREAD

**Discussion by wdduncan** [EvgSkv/logica#214] [Ideas]
Do you have an plans on implementing string functions for sqlite databases?
E.g.:
```
select replace('foo bar', 'bar', 'baz')
->
foo baz
```

**Comment by EvgSkv** [comment]
Hello @wdduncan 
`replace` function appears to be working in Logica with SQLite, it just needs to be camelcased, i.e. first letter capitalized:
```
$ python3 -m logica - run Q <<<'@Engine("sqlite"); Q(x) :- x == Replace("foo bar", "bar", "baz")'
+---------+
| col0    |
+---------+
| foo baz |
+---------+
```

Does it work for you?
Let me know if you encounter any problems.

**Comment by wdduncan** [comment]
Yes. That works. Thanks!

**Comment by wdduncan** [comment]
Another question ...  
Are you planning to implement REGEXP_REPLACE? E.g.:
```
logica - run Q <<<'@Engine("sqlite"); Q(x) :- x == RegexpReplace("foo bar", "o", "")'  
->
sqlite3.OperationalError: no such function: REGEXP_REPLACE
```

**Reply by EvgSkv** [reply]
I looked up and it appears that there is no function with functionality of `REGEXP_REPLACE` in SQLite itself. A usual work-around that we use in this case is implementing it in Python. [Here is an example of implementing Split function.](https://github.com/EvgSkv/logica/commit/5b8b68d717a8e16a293e41c8801c8df170120e1c)

There is no plan in-sight to systematically implement all BigQuery functions missing in SQLite, but you are welcome to send a PR implementing `REGEXP_REPLACE`. Furthermore if this blocks your usecase and you don't know how to implement it, let me know, I think I can do it quite soon.

**Comment by wdduncan** [comment]
Hi @EvgSkv  Thanks for the pointer to the `Split` function. Yes, you are right. `REGEXP_REPALCE` is not native the sqlite. It is an extension. For now the `Replace` function fits my purposes. `REGEXP_REPALCE` would just be a nice addition.
